### PR TITLE
py-bluepyefe, py-igor2: new version, new package igor2

### PR DESCRIPTION
**Pull request by matz-e**

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
FYI. Confirmed the new version sha256 and `py-igor2`'s homepage.

File: var/spack/repos/builtin/packages/py-igor2/package.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+from spack.package import *
+
+
+class PyIgor2(PythonPackage):
+    """igor2: interface for reading binary IGOR files."""
+
+    # pypi only has no sdist
+    homepage = "https://github.com/AFM-analysis/igor2"
+    url = "https://pypi.io/packages/py3/i/igor2/igor2-0.5.3-py3-none-any.whl"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-igor2/package.py:14]: Could also use the GitHub tarball

File: var/spack/repos/builtin/packages/py-igor2/package.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    license("LGPL-3.0-or-later")
+
+    version(
+        "0.5.3",
+        sha256="bb7b54a5926ec640e0e9176f46e0dd88ad956fec2d17ba3b0a7687eba82cefee",
+        expand=False,
+    )
+
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-igor2/package.py]: If you do stick with the wheel, setuptools isn't needed

File: var/spack/repos/builtin/packages/py-igor2/package.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    depends_on("py-numpy@1.25.1:", type=("build", "run"))
+    depends_on("py-matplotlib", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-igor2/package.py]: This is an optional dep, I would remove it or add a variant

File: var/spack/repos/builtin/packages/py-igor2/package.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    url = "https://pypi.io/packages/py3/i/igor2/igor2-0.5.3-py3-none-any.whl"
+
+    license("LGPL-3.0-or-later")
+
+    version(
+        "0.5.3",
+        sha256="bb7b54a5926ec640e0e9176f46e0dd88ad956fec2d17ba3b0a7687eba82cefee",
+        expand=False,
+    )
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-igor2/package.py:23]: Needs:
```python
depends_on("python@3.8:3", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-quantities/package.py [hunk 5]
```diff
@@ -15,6 +15,7 @@ class PyQuantities(PythonPackage):
 
     license("DOC")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-quantities/package.py]: Seems like an unrelated change, could be in a separate PR

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-quantities/package.py (hunk 5 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-quantities/package.py]: Sorry, but it in here as these three package updates are needed to make `bluepyefe` work with Python 3.11

File: var/spack/repos/builtin/packages/py-igor2/package.py (hunk 1 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-igor2/package.py:14]: `pyproject.toml` refers to `setuptools-scm` which requires info not present in the automatically generated tarballs?

File: var/spack/repos/builtin/packages/py-igor2/package.py (hunk 2 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-igor2/package.py]: OK, thanks.

File: var/spack/repos/builtin/packages/py-igor2/package.py (hunk 3 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-igor2/package.py]: Removed. Per `pyproject.toml`, `matplotlib` is needed for the CLI, but the CLI scripts seem to not have been updated after the fork from vanilla `igor`.

File: var/spack/repos/builtin/packages/py-quantities/package.py (hunk 5 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-quantities/package.py]: Can you elaborate? Does one of these packages not support Python 3.11?

File: var/spack/repos/builtin/packages/py-igor2/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-igor2/package.py:14]: Ah, I'm able to reproduce this issue. Might be worth opening an issue with the devs and adding a comment to explain why GitHub sdist isn't used either.

File: var/spack/repos/builtin/packages/py-quantities/package.py (hunk 5 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-quantities/package.py]: Hm, just noticed that this package dropped the requirements on Python. I forward-ported this PR from Spack 0.21, where Python was constrained to 3.10: https://github.com/spack/spack/blob/v0.21.1/var/spack/repos/builtin/packages/py-quantities/package.py#L23

I can take this package out of the PR, then.

File: var/spack/repos/builtin/packages/py-igor2/package.py (hunk 1 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-igor2/package.py:14]: [Looks like the next version will have a proper sdist on PyPI](https://github.com/AFM-analysis/igor2/commit/80c5612278cfee423b9802e1f7390743495941dd#diff-e77fed9000f2321f23f2709521ece8522ae833f771a39a7599275676d4ebe6aeR24). Do you still want a comment?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-igor2/package.py:14]: Nah, it's fine

File: var/spack/repos/builtin/packages/py-quantities/package.py (hunk 5 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-quantities/package.py]: And done.

**Review by adamjstewart — APPROVED** [review]
